tests/extract/conftest.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+from dotenv import load_dotenv
+import os
+from uuid import uuid4
+from datagovharvester.utils.s3_utilities import create_s3_client, create_s3_bucket
+
+
+@pytest.fixture
+def get_dcatus_job():
+    return {
+        "url": "http://localhost/data.json",
+        "source_id": str(uuid4()),
+        "job_id": str(uuid4()),
+    }
+
+
+@pytest.fixture
+def create_client_config():
+    config = {}
+    load_dotenv()
+
+    config["aws_access_key_id"] = os.getenv("S3FILESTORE_AWS_ACCESS_KEY_ID")
+    config["aws_secret_access_key"] = os.getenv("S3FILESTORE_AWS_SECRET_ACCESS_KEY")
+    config["region_name"] = os.getenv("S3FILESTORE_REGION_NAME")
+    config["endpoint_url"] = os.getenv("S3FILESTORE_HOST_NAME")
+
+    return config
+
+
+@pytest.fixture
+def create_client(create_client_config):
+    return create_s3_client(create_client_config)
+
+
+@pytest.fixture
+def create_bucket(create_client):
+    s3_client, s3_client_msg = create_client
+    return create_s3_bucket(s3_client, "test-bucket")
```

Comment:
We'll need to add tests after this. In the future we will follow [Railway Oriented Programming](https://swlaschin.gitbooks.io/fsharpforfunandprofit/content/posts/recipe-part2.html), in thinking this way we want to capture make sure each "track" (ie every exception) is working as we expect. This should include the following use cases:

1. request error
2. request doesn't give a 200 response code
3. request is not a valid JSON
4. record has no `dataset` key
5. record does not have an `id` (not positive if this is a warning or an error; it won't continue past compare process but maybe we still want to save it's state? I lean towards not saving it, this is a really edge case)
6. s3 upload failure
7. Happy path (record is good and is processed successfully)